### ISSUE TITLE
Nested schemas and rules

### ISSUE THREAD

**Issue by theozaurus** [dry-rb/dry-validation#214]
Using dry-validation 0.9.2 errors from rules no longer nest correctly:

```ruby
NestedSchema = Dry::Validation.Schema {
  required(:user).schema do
    required(:email).filled
    required(:password).filled
    required(:password_confirmation).filled

    rule(passwords_match: [:password, :password_confirmation]) { |p1, p2| p1.eql?(p2) }
  end
}

result = NestedSchema.call(user: {password: 'foo', password_confirmation: 'bar'})

expect(result.messages.fetch(:user).fetch(:email)).to eq(['is missing']) # => passes
expect(result.messages.fetch(:user).fetch(:password)).to eq(['must be equal to bar']) # => fails
```

**Comment by theozaurus** [comment]
Brilliant!

**Comment by solnic** [comment]
Fixed in 0.9.3

**Comment by theozaurus** [comment]
Thanks.
